### DISCUSSION TITLE
Laravel reverb

### DISCUSSION THREAD

**Discussion by AbhilashSivaraman** [laravel/framework#52250] [General]
I just implement in laravel reverb in local its works fine when  i goes to my remote environment i face such issues like
event is triggered successfully and connection is also connect successfully but  event and channels are not connect

Here my code

### In my web.php
 
```
Route::get('/', function () {
    Reverbevent::dispatch();

   return view('welcome');
});
```

### In app.js 

 To detect 
 
```
 import './bootstrap';



document.addEventListener('DOMContentLoaded',(e)=>{
    Echo.channel('reverb-channel').listen('Reverbevent',(e)=>{
         console.log(e);
        // alert('hiii');
    });
});

```
### Here my  event
```
<?php

namespace App\Events;

use Illuminate\Broadcasting\Channel;
use Illuminate\Broadcasting\InteractsWithSockets;
use Illuminate\Broadcasting\PresenceChannel;
use Illuminate\Broadcasting\PrivateChannel;
use Illuminate\Contracts\Broadcasting\ShouldBroadcast;
use Illuminate\Foundation\Events\Dispatchable;
use Illuminate\Queue\SerializesModels;

class Reverbevent implements ShouldBroadcast
{
    use Dispatchable, InteractsWithSockets, SerializesModels;

    /**
     * Create a new event instance.
     */
    public $sample="1233";
    public function __construct( )
    {
        //
        // $this->sample=$sample;
    }

    /**
     * Get the channels the event should broadcast on.
     *
     * @return array<int, \Illuminate\Broadcasting\Channel>
     */
    public function broadcastOn(): array
    {
        return [
            new Channel('reverb-channel'),
        ];
    }
}

```
### In channel.php

    ```
<?php

    use Illuminate\Support\Facades\Broadcast;

    
    Broadcast::channel('reverb-channel', function ($user, $id) {
        return true;
    });
```    
###  IN env 

i just include

QUEUE_CONNECTION=database

### and in network

![Screenshot from 2024-07-24 16-58-14](https://github.com/user-attachments/assets/62eab02a-cc70-43bf-a614-8e4d721b05c5)
![Screenshot from 2024-07-24 17-03-57](https://github.com/user-attachments/assets/d6e68e3b-bc59-4dde-84fe-7f51bb819d19)
in console

![Screenshot from 2024-07-24 17-08-17](https://github.com/user-attachments/assets/d0c140ec-6a9e-4056-b893-ef8dfa326a06)

**Comment by ImSidow** [comment]
I faced the same issue before. Make sure to double-check your REVERB_{HOST, PORT, SCHEME} environment variables. You might also need to add a channel route like this:

`
Broadcast::channel('user.{id}', function (User $user, string $id) {
    return (int) $user->id === (int) $id;
});
`
Best of luck

**Comment by murilolivorato** [comment]
@AbhilashSivaraman 
I  wrote a laravel reverb article in medium .
have a look -
https://medium.com/@murilolivorato/real-time-notifications-with-laravel-reverb-vue-3-laravel-echo-fully-dockerized-68eab58d5fc7
and I am sharing the file 

https://github.com/murilolivorato/real-time-notifications-laravel-reverb-vue
maybe it can help you
